pkg/agent/agent.go

```diff
@@ -71,8 +71,13 @@ func (a *Agent) ID() string {
 // readPump handles messages coming from the peer
 func (a *Agent) readPump() {
 	a.broker.Register(a.room, a)
+	a.broker.SendControl(a.room, messaging.NewPeerConnected(a.ID()))
+
 	defer func() {
+		close(a.writeChan)
```

Comment:
Shouldn't we move it after a.boker.Unregister? If we close the channel, but in the meantime another groutine tries to send something to it (the closed channel is still registered in the broker), the program will panic.